Data/ByteString/Builder/RealFloat/Internal.hs

```diff
@@ -467,7 +467,13 @@ timesWord2 a b =
   let ra = raw a
       rb = raw b
 #if WORD_SIZE_IN_BITS >= 64
+#if __GLASGOW_HASKELL__ < 903
       !(# hi, lo #) = ra `timesWord2#` rb
+#else
+      !(# hi_, lo_ #) = word64ToWord# ra `timesWord2#` word64ToWord# rb
+      hi = wordToWord64# hi_
+      lo = wordToWord64# lo_
```

Comment:
Exactly :(

Reply:
Might be worth reporting on the GHC issue tracker, and referencing the ticket in a comment here.

Reply:
It's likely impossible: a platform-dependent `timesWord2# :: Word# -> Word# -> (# Word#, Word# #)` is always a primop, but a hypothetical `timesWord642# :: Word64# -> Word64# -> (# Word64#, Word64# #)` is not, because it requires software emulation on 32-bit machines.

Reply:
`wordToWord64#` is a no-op on 64-bit machine and should not impose a performance penalty, we just lose a bit in readability.

Reply:
> but a hypothetical timesWord642# :: Word64# -> Word64# -> (# Word64#, Word64# #) is not, because it requires software emulation on 32-bit machines.

Yes but it's already the case for some operations: we fallback to the C implementation on 32-bit architectures. So we could definitely add `timesWord642#` (probably not with this name though).